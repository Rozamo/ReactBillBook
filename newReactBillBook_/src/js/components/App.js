import React from "react";
import SideBarItem from "./sidebar/SidebarItem.js";
import Content from "./Content.js";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";

class App extends React.Component {
	constructor(props) {
		super(props);
		this.state = {
			sidebarChoice: "invoices",
		};
	}

	changeSidebarChoice = (sidebarChoice) => {
		this.setState({
			sidebarChoice,
		});
	};

	render() {
		const arr = ["Customers", "Items", "Invoices"];
		const sidebarItemStyle = {
			textDecoration: "none",
			color: "black",
		};

		return (
			<div className="outside-box">
				<Router>
					<div className="side-bar">
						{arr.map((el, index) => (
							<Link
								key={index}
								style={sidebarItemStyle}
								to={"/" + el.toLowerCase()}
							>
								<SideBarItem
									value={el}
									sidebarChoice={el.toLowerCase()}
									cls={el
										.toLowerCase()
										.concat(
											this.state.sidebarChoice === el.toLowerCase()
												? " active"
												: ""
										)}
									changeSidebarChoice={this.changeSidebarChoice}
								/>
							</Link>
						))}
					</div>

					<Switch>
						<Route
							exact
							path="/"
							render={() => <Content sidebarChoice="invoices" />}
						></Route>
						<Route
							exact
							path="/invoices"
							render={() => <Content sidebarChoice="invoices" />}
						></Route>
						<Route
							exact
							path="/customers"
							render={() => <Content sidebarChoice="customers" />}
						></Route>
						<Route
							exact
							path="/items"
							render={() => <Content sidebarChoice="items" />}
						></Route>
					</Switch>
				</Router>
			</div>
		);
	}
}

export default App;
