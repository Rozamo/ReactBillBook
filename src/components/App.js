import React, { useEffect, useState } from 'react';
import { BrowserRouter as Router, Switch, Route, Link } from 'react-router-dom';
import SidebarItem from './sidebar/SidebarItem';
import CustomersTable from './customers/CustomersTable';
import CustomersForm from './customers/CustomersForm';
import ItemsTable from './items/ItemsTable';
import ItemsForm from './items/ItemsForm';
import Invoices from './invoices/Invoices';

export default function App() {
  const param = window.location.href;
  const [sidebarChoice, setSidebarChoice] = useState(() => {
    if (param.includes('customers'))
      return 'customers';
    else if (param.includes('items'))
      return 'items';
    else if (param.includes('invoices'))
      return 'invoices';
    else
      return '';
  });
  function changeSidebarChoice(sidebarChoice) {
    setSidebarChoice(sidebarChoice);
  }
  // useEffect(() => {
  //     console.log(sidebarChoice);
  // });
  return (
    <Router>
      <div className="outside-box">
        <div className="side-bar">
          <Link to='/customers/list'>
            <SidebarItem value="Customers" sidebarChoice="customers" classValue={"customers".concat(sidebarChoice === "customers" ? " active" : "")} changeSidebarChoice={changeSidebarChoice} />
          </Link>
          <Link to='/items/list'>
            <SidebarItem value="Items" sidebarChoice="items" classValue={"items".concat(sidebarChoice === "items" ? " active" : "")} changeSidebarChoice={changeSidebarChoice} />
          </Link>
          <Link to='/invoices'>
            <SidebarItem value="Invoices" sidebarChoice="invoices" classValue={"invoices".concat(sidebarChoice === "invoices" ? " active" : "")} changeSidebarChoice={changeSidebarChoice} />
          </Link>
        </div>
        <Switch>
          <Route exact path='/' component={() => <div className="content"></div>} />
          <Route path='/customers/list' component={CustomersTable} />
          <Route path='/customers/create' component={CustomersForm} />
          <Route path='/items/list' component={ItemsTable} />
          <Route path='/items/create' component={ItemsForm} />
          <Route exact path="/invoices" component={Invoices}></Route>
        </Switch>
      </div>
    </Router>
  );
}
