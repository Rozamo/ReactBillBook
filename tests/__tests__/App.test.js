import React from "react";
import {
  render,
  fireEvent,
  screen,
  waitFor,
  cleanup,
} from "@testing-library/react";
import { BrowserRouter as Router, useHistory } from "react-router-dom";
import App from "../../src/components/App";

const fakeData = {
  entity: "collection",
  items: [
    {
      name: "Jmp",
      email: "J@rzp.com",
      age: "32",
      address: "123, Charming Avenue",
      amount: 1231,
    },
  ],
};

test("should render customer list and form inside App", async () => {
  global.fetch = jest.fn().mockImplementation(() =>
    Promise.resolve({
      json: () => new Promise((resolve) => resolve(fakeData)),
    })
  );

  const { container } = render(
    <Router>
      <App />
    </Router>
  );

  expect(container.textContent).toContain("Customers");
  expect(container.textContent).toContain("Items");
  expect(container.textContent).toContain("Invoices");
  expect(container).toMatchSnapshot();

  const custBtn = container.getElementsByClassName("customers")[0];
  const itemsBtn = container.getElementsByClassName("items")[0];
  const invoicesBtn = container.getElementsByClassName("invoices")[0];

  expect(custBtn.textContent).toContain("Customers");
  expect(itemsBtn.textContent).toContain("Items");
  expect(invoicesBtn.textContent).toContain("Invoices");

  expect(custBtn.classList.contains("active")).toBeFalsy();
  expect(itemsBtn.classList.contains("active")).toBeFalsy();
  expect(invoicesBtn.classList.contains("active")).toBeFalsy();

  // Click on customers tab
  fireEvent.click(custBtn);

  await waitFor(() => container);

  expect(custBtn.classList.contains("active")).toBeTruthy();
  expect(itemsBtn.classList.contains("active")).toBeFalsy();
  expect(invoicesBtn.classList.contains("active")).toBeFalsy();

  expect(container).toMatchSnapshot();
  expect(container.querySelector(".content").textContent).toContain(
    "Customers"
  );

  const newCustBtn = container.querySelector(".content #button");

  fireEvent.click(newCustBtn);

  await waitFor(() => container);

  expect(container.querySelector(".content").textContent).toContain("Email");
  expect(container.querySelectorAll(".content input").length).toBe(3);
  expect(container).toMatchSnapshot();

  // CLick on items tab
  fireEvent.click(itemsBtn);

  await waitFor(() => container);

  expect(custBtn.classList.contains("active")).toBeFalsy();
  expect(itemsBtn.classList.contains("active")).toBeTruthy();
  expect(invoicesBtn.classList.contains("active")).toBeFalsy();

  expect(container).toMatchSnapshot();
  expect(container.querySelector(".content").textContent).toContain("Items");

  const newItemBtn = container.querySelector(".content #button");
  fireEvent.click(newItemBtn);

  await waitFor(() => container);

  expect(container.querySelector(".content").textContent).toContain(
    "Description"
  );
  expect(container.querySelectorAll(".content input").length).toBe(2);
  expect(container).toMatchSnapshot();

  // Click on invoices tab

  fireEvent.click(invoicesBtn);

  await waitFor(() => container);

  expect(custBtn.classList.contains("active")).toBeFalsy();
  expect(itemsBtn.classList.contains("active")).toBeFalsy();
  expect(invoicesBtn.classList.contains("active")).toBeTruthy();

  expect(container).toMatchSnapshot();
  expect(container.querySelector(".content").textContent).toContain("Invoices");

  const newInvoiceBtn = container.querySelector(".content #button");
  fireEvent.click(newInvoiceBtn);

  await waitFor(() => container);

  expect(container.querySelector(".content").textContent).toContain("Bill to");
  expect(container.querySelectorAll("input[type='date']").length).toBe(2);
  expect(container).toMatchSnapshot();

  global.fetch.mockClear();
  delete global.fetch;
});
