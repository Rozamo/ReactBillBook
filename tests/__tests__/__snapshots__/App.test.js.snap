// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render customer list and form inside App 1`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    />
  </div>
</div>
`;

exports[`should render customer list and form inside App 2`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers active"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="top-panel"
      >
        <h1
          id="title"
        >
          Customers
        </h1>
        <a
          href="/customers/create"
        >
          <button
            id="button"
            type="button"
            value=""
          >
            <img
              alt="New"
              id="floppy"
              src="test-file-stub"
            />
            New Customer
          </button>
        </a>
      </div>
      <table
        class="inv-table"
        id="inv-table"
      >
        <thead>
          <tr>
            <th>
              NAME
            </th>
            <th>
              PHONE
            </th>
            <th>
              EMAIL
            </th>
            <th>
              CREATED ON
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class=""
            >
              Jmp
            </td>
            <td
              class=""
            />
            <td
              class=""
            >
              J@rzp.com
            </td>
            <td
              class=""
            >
              NaN Invalid Date NaN
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`should render customer list and form inside App 3`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers active"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="top-panel"
      >
        <h1
          id="title"
        >
          New Customer
        </h1>
      </div>
      <form
        class="customer-form"
      >
        <div
          class="cust-panel-1"
        >
          <div>
            <label
              for="name"
            >
              Name
            </label>
            <input
              name="name"
              required=""
              type="text"
              value=""
            />
          </div>
          <div>
            <label
              for="contact"
            >
              Phone
            </label>
            <input
              name="contact"
              type="text"
              value=""
            />
          </div>
        </div>
        <label
          for="email"
        >
          Email
        </label>
        <div
          class="cust-panel-1"
        >
          <input
            name="email"
            type="email"
            value=""
          />
          <button
            id="button"
            type="submit"
            value=""
          >
            <img
              alt="Save"
              id="floppy"
              src="test-file-stub"
            />
            Save Customer
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`should render customer list and form inside App 4`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items active"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="top-panel"
      >
        <h1
          id="title"
        >
          Items
        </h1>
        <a
          href="/items/create"
        >
          <button
            id="button"
            type="button"
            value=""
          >
            <img
              alt="New"
              id="floppy"
              src="test-file-stub"
            />
            Add Item
          </button>
        </a>
      </div>
      <table
        class="inv-table"
        id="inv-table"
      >
        <thead>
          <tr>
            <th>
              NAME
            </th>
            <th>
              DESCRIPTION
            </th>
            <th>
              PRICE
            </th>
            <th>
              ADDED ON
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td
              class=""
            >
              Jmp
            </td>
            <td
              class=""
            />
            <td
              class=""
            >
              ₹12.31
            </td>
            <td
              class=""
            >
              NaN Invalid Date NaN
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`should render customer list and form inside App 5`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items active"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="top-panel"
      >
        <h1
          id="title"
        >
          New Item
        </h1>
      </div>
      <form
        class="customer-form"
        style="width: 50%;"
      >
        <label
          for="name"
        >
          Name
        </label>
        <input
          name="name"
          type="text"
          value=""
        />
        <label
          for="amount"
        >
          Price
        </label>
        <input
          name="amount"
          type="number"
          value=""
        />
        <label
          for="name"
        >
          Description
        </label>
        <textarea
          name="description"
          type="text"
        />
        <br />
        <button
          id="button"
          type="submit"
          value=""
        >
          <img
            alt="Save"
            id="floppy"
            src="test-file-stub"
          />
          Save Item
        </button>
      </form>
    </div>
  </div>
</div>
`;

exports[`should render customer list and form inside App 6`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices active"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="top-panel"
      >
        <h1>
          Invoices
        </h1>
        <a
          href="/invoices/create"
        >
          <button
            id="button"
            value="create-invoice"
          >
            <img
              alt="Save"
              id="floppy"
              src="test-file-stub"
            />
            New Invoice
          </button>
        </a>
      </div>
      <table
        class="inv-table"
        id="inv-table"
      >
        <tbody>
          <tr>
            <th>
              DATE
            </th>
            <th>
              CUSTOMER
            </th>
            <th>
              PAID STATUS
            </th>
            <th>
              AMOUNT
            </th>
            <th>
              AMOUNT DUE
            </th>
          </tr>
          <tr>
            <td>
              NaN Invalid Date NaN
            </td>
            <td />
            <td>
              <mark>
                 
                 
              </mark>
            </td>
            <td>
              ₹12.31
            </td>
            <td>
              0
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>
</div>
`;

exports[`should render customer list and form inside App 7`] = `
<div>
  <div
    class="outside-box"
  >
    <div
      class="side-bar"
    >
      <a
        href="/customers/list"
      >
        <div
          class="customers"
        >
          <svg
            class="bi bi-person-fill"
            fill="currentColor"
            height="1em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 16 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3 14s-1 0-1-1 1-4 6-4 6 3 6 4-1 1-1 1H3zm5-6a3 3 0 1 0 0-6 3 3 0 0 0 0 6z"
              fill-rule="evenodd"
            />
          </svg>
          Customers
        </div>
      </a>
      <a
        href="/items/list"
      >
        <div
          class="items"
        >
          <svg
            class="bi bi-star-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 17px;"
            viewBox="0 0 16 16"
            width="0.9em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M3.612 15.443c-.386.198-.824-.149-.746-.592l.83-4.73L.173 6.765c-.329-.314-.158-.888.283-.95l4.898-.696L7.538.792c.197-.39.73-.39.927 0l2.184 4.327 4.898.696c.441.062.612.636.283.95l-3.523 3.356.83 4.73c.078.443-.36.79-.746.592L8 13.187l-4.389 2.256z"
            />
          </svg>
          Items
        </div>
      </a>
      <a
        href="/invoices"
      >
        <div
          class="invoices active"
        >
          <svg
            class="bi bi-file-earmark-text-fill"
            fill="currentColor"
            height="0.9em"
            style="margin: 0px 25px 0px 15px;"
            viewBox="0 0 14 16"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M2 2a2 2 0 0 1 2-2h5.293A1 1 0 0 1 10 .293L13.707 4a1 1 0 0 1 .293.707V14a2 2 0 0 1-2 2H4a2 2 0 0 1-2-2V2zm7 2l.5-2.5 3 3L10 5a1 1 0 0 1-1-1zM4.5 8a.5.5 0 0 0 0 1h7a.5.5 0 0 0 0-1h-7zM4 10.5a.5.5 0 0 1 .5-.5h7a.5.5 0 0 1 0 1h-7a.5.5 0 0 1-.5-.5zm0 2a.5.5 0 0 1 .5-.5h4a.5.5 0 0 1 0 1h-4a.5.5 0 0 1-.5-.5z"
              fill-rule="evenodd"
            />
          </svg>
          Invoices
        </div>
      </a>
    </div>
    <div
      class="content"
    >
      <div
        class="top-panel"
      >
        <h1>
          New Invoice
        </h1>
        <button
          id="button"
          value="create-invoice"
        >
          <img
            alt="Save"
            id="floppy"
            src="test-file-stub"
          />
          Save Invoice
        </button>
      </div>
      <div
        class="panel-1"
      >
        <div
          class="inv-cust-panel"
        >
          <div
            style="color: rgb(125, 123, 123);"
          >
            Bill to
          </div>
          <div
            class="inv-cust-det"
          >
            <div>
              No Customer Chosen
            </div>
            <button
              id="inv-cust-change"
            >
              Choose
            </button>
          </div>
        </div>
        <div>
          <label>
            Issued at
          </label>
          <input
            id="issued-at"
            name="issued-at"
            style="width: 170px;"
            type="date"
          />
        </div>
        <div>
          <label>
            Due-Date
          </label>
          <input
            id="due-date"
            name="due-date"
            style="width: 170px;"
            type="date"
          />
        </div>
      </div>
      <div
        class="inv-item-panel"
      >
        <table
          class="item-tbl"
          id="inv-item-tbl"
        >
          <tbody>
            <tr />
          </tbody>
        </table>
        <div
          class="item-btn"
        >
          <button
            id="add-item"
          >
            <svg
              class="bi bi-basket2-fill"
              fill="currentColor"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5.929 1.757a.5.5 0 1 0-.858-.514L2.217 6H.5a.5.5 0 0 0-.5.5v1a.5.5 0 0 0 .5.5h.623l1.844 6.456A.75.75 0 0 0 3.69 15h8.622a.75.75 0 0 0 .722-.544L14.877 8h.623a.5.5 0 0 0 .5-.5v-1a.5.5 0 0 0-.5-.5h-1.717L10.93 1.243a.5.5 0 1 0-.858.514L12.617 6H3.383L5.93 1.757zM4 10a1 1 0 0 1 2 0v2a1 1 0 1 1-2 0v-2zm3 0a1 1 0 0 1 2 0v2a1 1 0 1 1-2 0v-2zm4-1a1 1 0 0 0-1 1v2a1 1 0 1 0 2 0v-2a1 1 0 0 0-1-1z"
                fill-rule="evenodd"
              />
            </svg>
            Add Item
          </button>
        </div>
      </div>
      <hr />
      <div
        class="panel-2"
      >
        <div
          class="notes"
        >
          <div
            class="inv-description"
          >
            <textarea
              cols="30"
              id="description"
              placeholder="Write the description here"
              rows="10"
            />
          </div>
        </div>
        <div
          class="item-total"
          id="inv-items-list"
        >
          <table
            class="item-total-tbl"
          >
            <tbody />
          </table>
          <span />
          <div
            class="item-total-last-row"
          >
            <span>
              TOTAL AMOUNT:
            </span>
            <span
              id="item-total-amount"
            >
              0
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
